reframe/frontend/cli.py

```diff
@@ -903,13 +903,15 @@ def module_unuse(*paths):
         exec_policy.sched_flex_alloc_nodes = sched_flex_alloc_nodes
         parsed_job_options = []
         for opt in options.job_options:
-            optstr, *_ = re.split(r'=|\s+', opt)
+            opt_split = re.split(r'=|\s+', opt, maxsplit=1)
+            optstr = opt_split[0]
+            valstr = opt_split[1] if len(opt_split) > 1 else ''
             if opt.startswith('-') or opt.startswith('#'):
                 parsed_job_options.append(opt)
             elif len(optstr) == 1:
-                parsed_job_options.append(f'-{opt}')
+                parsed_job_options.append(f'-{optstr} {valstr}')
             else:
-                parsed_job_options.append(f'--{opt}')
+                parsed_job_options.append(f'--{optstr} {valstr}')
```

Comment:
```suggestion
                parsed_job_options.append(f'--{optstr}={valstr}')
```

Reply:
Here I left out the `=` on the long options too to cover cases without like `-J foo` without having to do anything extra. Are `--opt=foo` and `--opt foo` equivalent?

Reply:
Good point about the `-J foo`, so you can can leave it as you have it.

> Are --opt=foo and --opt foo equivalent?

Yes, they are.